audio-worklet/src/bitcrusher-controller.js

```diff
@@ -0,0 +1,145 @@
+in the documentation and/or other materials provided with the
+distribution.
+    * Neither the name of Google Inc. nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+/**
+ *  Loads sound resources and initiates and configures a Web Audio graph given
+ * GUI controls specified in bitcrusher.html. 
+ * Sound is routed through a bitcrusher wrapper class defined in bitcrusher.js
+ */
+
+/**-----------------------------------------------------------------------------
+ 	Initialize variables and retrieve song from server
+   --------------------------------------------------------------------------**/
+let context = new AudioContext();
+let bitDepth = 8.0; // the precision in bits of the final sample
+let reduction = 10; // bitcrushed buffer will have [reduction] equal samples 
+
+let freq = 440; // default frequency to A4
+let oscillator; // oscillator node
+let mp3Source; // mp3 sound source
+let mp3Buffer; // the Audio Buffer of loaded mp3 track
+let bitcrusher; // script processor wrapper implementing a bit crusher node
+const songs = ["lonely-tentacle.mp3", "revenge.mp3", "it-from-bit.mp3"];
+const folder = "audio/";
+const songUrl = folder + songs[2]; // specified url will be loaded
+
+
+// Retrieve local bitcrusher mp3 file and decode as audio buffer
+fetch(songUrl).then(function(response) { 
+	return response.arrayBuffer();}).then(function(song) { 
+		context.decodeAudioData(song, function(buffer) { // decode as audio buffer
+  		mp3Buffer = buffer;
+    	document.getElementById("startMp3").disabled = false; // enable mp3 button
+  }, function (d) { console.log(d)});
+});
+
+
+/**-----------------------------------------------------------------------------
+***GUI Controls: modify bitcrusher parameters
+***---------------------------------------------------------------------------*/
+
+/**
+ * When oscillator button is clicked, run bitcrushed oscillator at
+ * frequency, bit depth, and sample rate reduction specified in GUI 
+ */
+function testOscillator() {
+  oscillator = new OscillatorNode(context); 
+  bitcrusher = new Bitcrusher(context, {buffersize: 512, inputChannels: 1, 
```

Comment:
512 doesn't work very well on Linux.  And for many common Android phones, it sound really horrible.  Let's use a buffer size of 0 so that Chrome chooses something somewhat reasonable.  We're not looking for best latency in this example.